src/services/identity/LaunchClient.ts

```diff
@@ -35,21 +40,130 @@ class LaunchClient {
+  }
+
+  /**
+   * The rate limit for Create Domain Association is 10 per hour.
+   * We want to limit interference with operations, as such we mock this call during development.
+   * @returns Mocked output for CreateDomainAssociationCommand
+   */
+  mockCreateDomainAssociationOutput = (
+    input: CreateDomainAssociationCommandInput
+  ): Promise<CreateDomainAssociationCommandOutput> => {
+    const mockResponse: CreateDomainAssociationCommandOutput = {
+      $metadata: {
+        httpStatusCode: 200,
+      },
+      domainAssociation: {
+        autoSubDomainCreationPatterns: [],
+        autoSubDomainIAMRole: undefined,
+        certificateVerificationDNSRecord: undefined,
+        domainAssociationArn: `arn:aws:amplify:ap-southeast-1:11111:apps/${input.appId}/domains/${input.domainName}`,
+        domainName: input.domainName,
+        domainStatus: "CREATING",
+        enableAutoSubDomain: false,
+        statusReason: undefined,
+        subDomains: undefined,
+      },
+    }
+
+    const subDomainSettingsList = input.subDomainSettings
+    if (!subDomainSettingsList || subDomainSettingsList.length === 0) {
+      return Promise.resolve(mockResponse)
+    }
+
+    const subDomains: SubDomain[] = this.getSubDomains(subDomainSettingsList)
+
+    // We know that domainAssociation is not undefined, so we can use the non-null assertion operator
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    mockResponse.domainAssociation!.subDomains = subDomains
+    return Promise.resolve(mockResponse)
+  }
+
+  private isTestEnv() {
+    return config.get("env") === "test" || config.get("env") === "dev"
+  }
+
+  private getSubDomains(subDomainList: SubDomainSetting[], hasCreated = false) {
+    const subDomainPrefixList = subDomainList
+      .map((subDomain) => subDomain.prefix)
+      .filter((prefix) => prefix !== undefined) as string[]
+
+    const subDomains: SubDomain[] = []
+    subDomainPrefixList.forEach((subDomainPrefix) => {
+      subDomains.push({
+        dnsRecord: `${subDomainPrefix} CNAME ${
+          hasCreated ? "test.cloudfront.net" : "<pending>"
+        }`,
+        subDomainSetting: {
+          branchName: "master",
+          prefix: subDomainPrefix,
+        },
+        verified: false,
+      })
+    })
+    return subDomains
+  }
+
+  mockGetDomainAssociationOutput(
+    input: MockGetDomainAssociationCommandInput
+  ): Promise<GetDomainAssociationCommandOutput> {
+    const mockResponse: GetDomainAssociationCommandOutput = {
+      $metadata: {
+        httpStatusCode: 200,
+      },
+      domainAssociation: {
+        certificateVerificationDNSRecord: `testcert.${input.domainName}. CNAME testcert.acm-validations.aws.`,
+        domainAssociationArn: `arn:aws:amplify:ap-southeast-1:11111:apps/${input.appId}/domains/${input.domainName}`,
+        domainName: input.domainName,
+        domainStatus: "PENDING_VERIFICATION",
```

Comment:
Is this the expected status? What's the terminal status? "Verified"?

Reply:
Yup, the moment we create the domain and get the required DNS values, we dont expect a success state at the moment. the full list of status can be found [here](https://repost.aws/questions/QUxsRQ8_2KSJ68j-Dk5M7Xvg/amplify-domain-creation-statuses-what-do-they-mean) are  
PENDING_VERIFICATION | IN_PROGRESS | AVAILABLE | PENDING_DEPLOYMENT | FAILED | CREATING | REQUESTING_CERTIFICATE | UPDATING

Reply:
Then do we return `AVAILABLE` or `VERIFIED`? I don't see those statuses in the mock response

Reply:
The sequence of events is 
1. Send CreateDomainAssociation Call <-- at this point we don't know the DNS records to set, Amplify still generating 
2. Send GetDomainAssociationCall <-- at this point we call this just to get the new DNS records to set 

At the later point, the call is not meant to be `Available` or `Verified`. This is not the part of the process where the user has already put in the DNS records. As such, to mimic the actual behaviour from Amplify, we use `PENDING_VERIFICATION`

Reply:
ok got it